main.nf

```diff
@@ -1036,9 +1076,117 @@ process merge_featureCounts {
+        } else if (reverseStranded) {
+            strandedness = 'SR'
+        }
+        def end_strandedness = params.singleEnd ? strandedness : "I${strandedness}"
+        def endedness = params.singleEnd ? "-r ${reads[0]}" : "-1 ${reads[0]} -2 ${reads[1]}"
+        """
+        salmon quant --validateMappings \\
+                        --seqBias --useVBOpt --gcBias \\
+                        --geneMap ${gtf} \\
+                        --threads ${task.cpus} \\
+                        --libType=${end_strandedness} \\
+                        --index ${index} \\
+                        $endedness \\
+                        -o ${sample}
+
+        # Replace first occurence of "TPM" from output .sf file with sample ID for easy merging
+        csvtk cut -t -f "-Length,-EffectiveLength,-NumReads" ${sample}/quant.sf \\
+            | sed "s:TPM:${sample}:" \\
+            > ${sample}/${sample}.quant.ids-only.txt
+
+        # Replace first occurence of "TPM" from output .sf file with sample ID for easy merging
+        csvtk cut -t -f "-Length,-EffectiveLength,-NumReads" ${sample}/quant.genes.sf \\
+            | sed "s:TPM:${sample}:" \\
+            > ${sample}/${sample}.quant.genes.ids-only.txt
+        """
+        }
+
+    process salmon_merge_transcript {
+      label 'low_memory'
+      publishDir "${params.outdir}/salmon", mode: 'copy'
+
+      input:
+      file transcript_quants from salmon_transcript_quant
+      file gtf from gtf_merge_salmon_quant.collect()
+
+      output:
+      file 'salmon_merged_transcript_tpm.csv'
+
+      script:
+      //if we only have 1 file, just use cat and pipe output to csvtk. Else join all files first, and then remove unwanted column names.
+      def single = transcript_quants instanceof Path ? 1 : transcript_quants.size()
+      def merge = (single == 1) ? 'cat' : 'csvtk join -t -f "Name"'
+      """
+      ## Merge transcript counts
+      ## Gene transcript_id <--> gene_name mapping
+      awk -F "\\t" '\$3 == "transcript" { print \$9 }' $gtf | grep -oP '(?<=transcript_id ")(\\w+)' > transcript_ids.txt
+      awk -F "\\t" '\$3 == "transcript" { print \$9 }' $gtf | grep -oP '(?<=gene_name ")(\\w+)' > transcript_gene_names.txt
+      paste transcript_ids.txt transcript_gene_names.txt > transcript_ids__to__gene_names.txt
+      $merge $transcript_quants \\
+        | csvtk join -t -f 1 transcript_ids__to__gene_names.txt - \\
+        | awk '{FS="\\t"; OFS="\\t"} { if (length(\$2) == 0) {\$1=\$1} else {\$1=\$2 " ("\$1")"}; \$2="" ; print \$0 }' \\
+        | cut  -f '1,3-' \\
+        | csvtk tab2csv \\
+        > salmon_merged_transcript_tpm.csv
+      """
+    }
+    process salmon_merge_gene {
+      label 'low_memory'
+      publishDir "${params.outdir}/salmon", mode: 'copy'
+
+      input:
+      file gene_quants from salmon_gene_quant
+      file featurecounts_merged from featurecounts_merged.collect() // Use gene_id > gene_name mapping from featurecounts to make sure it matches
```

Comment:
This is going to be a problem if we want to run salmon by itself because these channels wont be created. Do we have to do this step @olgabot @lpantano? Or is there an alternative?

Reply:
I have an alternative to merge files, but that probably will need another PR after this is merged. My idea is to use tximport package that is meant to do exactly that. And it won't depend only on salmon files. So the option is to remove this for now, and leave it not merged, and I can work after that in adding that step.

Reply:
How would you get the annotation via `tximport`? Would it be from the `gtf` file? 

You can create a PR relative to the `salmon` branch to add this functionality so we can get everything working here before we merge to `dev` 👍 

Would be good if you can incorporate the `rna_strandedness` comment below too for consistency with `HiSat2`.

Reply:
yes, I can do it to salmon if you wait for me :) I have implemented different strategies if only transcriptome is available, if gtf is available as well or if the user gives a full annotation genome name to get a fully annotated transcripts, genes, etc. I can work on that tomorrow.

Reply:
Ok. Ill fix some other stuff I have noticed too and merge that PR now so you can start adding in your bit.